test/runnable/imports/bar10378.d

```diff
@@ -0,0 +1,4 @@
+
+module bar;
```

Comment:
> That's actually a feature of D. The file name doesn't have to match the module name.

It's actually the result of a horribly broken implementation. The module is inserted w/ both names into the global symbol table.

> I'm not sure why the error is not reported

The error is only reported if the module is know under the module declaration name before importing it, this happens when it's parsed before the module importing it.